api/maven-api-core/src/main/java/org/apache/maven/api/Dependency.java

```diff
@@ -23,9 +23,9 @@
 public interface Dependency extends Artifact {
```

Comment:
Realizing now this inheritance hierarchy is wonky. This suggests a Dependency *is* an artifact, which isn't true. Nothing to do about that in this PR of course, but I wanted to call it out while I'm thinking about it.

Reply:
I think this is true in this context.  This is unrelated to the model's dependency list.
Here, the `Dependency` is the resolved `Artifact` that come from a `DependencyCoordinate`.  This is more related to the dependency collection and artifact resolution.  The `DependencyCoordinate` is what maps closely to the model's `Dependency`.